wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+        # basically need to override DoGetBestSize and AcceptsFocusFromKeyboard.
+
+        wx.Control.__init__(self, parent, id, pos, size, style, validator, name)
+
+        self.SYS_DEFAULT_GUI_FONT = wx.SystemSettings.GetFont(wx.SYS_DEFAULT_GUI_FONT)
+
+        # Initialize our cool bitmaps.
+        self.InitializeBitmaps()
+
+        # Initialize the focus pen colour/dashes, for faster drawing later.
+        self.InitializeColours()
+
+        # By default, we start unchecked.
+        self._checked = False
+
+        # Set the spacing between the check bitmap and the label to 3 by default.
+        # This can be changed using SetSpacing later.
+        self._spacing = 3
+        self._hasFocus = False
+
+        # Ok, set the wx.PyControl label, its initial size (formerly known an
+        # SetBestFittingSize), and inherit the attributes from the standard
+        # wx.CheckBox .
+        self.SetLabel(label)
+        self.SetInitialSize(size)
+        self.InheritAttributes()
+
+        # Bind the events related to our control: first of all, we use a
+        # combination of wx.BufferedPaintDC and an empty handler for
+        # wx.EVT_ERASE_BACKGROUND (see later) to reduce flicker.
+        self.Bind(wx.EVT_PAINT, self.OnPaint)
+        self.Bind(wx.EVT_ERASE_BACKGROUND, self.OnEraseBackground)
+
+        # Then we want to monitor user clicks, so that we can switch our
+        # state between checked and unchecked.
+        self.Bind(wx.EVT_LEFT_DOWN, self.OnMouseClick)
+        if wx.Platform == '__WXMSW__':
+            # MSW Sometimes does strange things...
+            self.Bind(wx.EVT_LEFT_DCLICK,  self.OnMouseClick)
+
+        # We want also to react to keyboard keys, namely the
+        # space bar that can toggle our checked state.
+        self.Bind(wx.EVT_KEY_UP, self.OnKeyUp)
+
+        # Then, we react to focus event, because we want to draw a small
+        # dotted rectangle around the text if we have focus.
+        # This might be improved!!!
+        self.Bind(wx.EVT_SET_FOCUS, self.OnSetFocus)
+        self.Bind(wx.EVT_KILL_FOCUS, self.OnKillFocus)
+
+
+    def InitializeBitmaps(self):
+        """ Initializes the check bitmaps. """
+
+        # We keep 4 bitmaps for GenCheckBox, depending on the
+        # checking state (Checked/UnChecked) and the control
+        # state (Enabled/Disabled).
+
+        self._bitmaps = {
+            "CheckedEnable": _GetCheckedBitmap(self),
+            "UnCheckedEnable": _GetNotCheckedBitmap(self),
+            "CheckedDisable": _GetCheckedImage(self).ConvertToDisabled().ConvertToBitmap(),
+            "UnCheckedDisable": _GetNotCheckedImage(self).ConvertToDisabled().ConvertToBitmap()}
+
+    def InitializeColours(self):
+        """ Initializes the focus indicator pen. """
+
+        textClr = self.GetForegroundColour()
+        if wx.Platform == "__WXMAC__":
+            self._focusIndPen = wx.Pen(textClr, 1, wx.PENSTYLE_SOLID)
+        else:
```

Comment:
Pens on Mac support the dashed style now so this isn't needed.

Reply:
removed these lines